pkg/policy/policy.go

```diff
@@ -0,0 +1,325 @@
+package policy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/starboard/pkg/kube"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	keyPrefixPolicy  = "policy."
+	keyPrefixLibrary = "library."
+	keySuffixKinds   = ".kinds"
+	keySuffixRego    = ".rego"
+)
+
+const (
+	kindAny      = "*"
+	kindWorkload = "Workload"
+)
+
+// Metadata describes policy metadata.
+type Metadata struct {
+	ID          string
+	Title       string
+	Severity    v1alpha1.Severity
+	Type        string
+	Description string
+}
+
+// NewMetadata constructs new Metadata based on raw value.
+func NewMetadata(value map[string]interface{}) (Metadata, error) {
+	severity, err := v1alpha1.StringToSeverity(fmt.Sprintf("%v", value["severity"]))
+	if err != nil {
+		return Metadata{}, fmt.Errorf("failed parsing severity from raw value: %w", err)
+	}
+	return Metadata{
+		Severity:    severity,
+		ID:          fmt.Sprintf("%v", value["id"]),
+		Title:       fmt.Sprintf("%v", value["title"]),
+		Type:        fmt.Sprintf("%v", value["type"]),
+		Description: fmt.Sprintf("%v", value["description"]),
+	}, nil
+}
+
+// Result describes result of evaluating a policy with a given input resource.
+type Result struct {
+	ID       string
+	Title    string
+	Severity v1alpha1.Severity
+	Type     string
+
+	Message string
+}
+
+// NewResult constructs new Result based on raw value.
+func NewResult(value map[string]interface{}) Result {
+	return Result{
```

Comment:
same here value["id"] should be checked for existance

Reply:
Done